### ISSUE TITLE
[SmartOS] Broken collectd-network in 2016Q4

### ISSUE THREAD

**Issue by hadret** [TritonDataCenter/pkgsrc-legacy#483]
Hi,

recently I noticed that nodes I migrated to base-64-lts 16.4.1 are no longer sending metrics to my central graphite server. Turned out that `collectd` daemon was failing to start due to fatal error. Here's the message:

```
[ Apr 13 11:03:47 Enabled. ]
[ Apr 13 11:03:47 Executing start method ("/opt/local/sbin/collectd"). ]
configfile: stat (/opt/local/etc/collectd.d/*.conf) failed: No such file or directory
ld.so.1: collectd: fatal: relocation error: file /opt/local/lib/collectd/network.so: symbol gcry_control: referenced symbol not found
[ Apr 13 11:03:47 Method "start" failed due to signal KILL. ]
```

image:base-64-lts 16.4.1
collectd:`5.6.2`
platform:`joyent_20161222T003450Z`

Regards,
Filip

**Comment by mamash** [comment]
Hm, it's strange. I see the network.so lib doesn't link libgcrypt like it should (I think), but I can't bring collectd crashing on the missing symbol, it just keeps running. I'll continue investigating.

**Comment by hadret** [comment]
Ah, probably I should have include also the configuration I'm using! Here's the relevant part:

```
LoadPlugin network
<Plugin network>
        <Server "$REMOTE_SERVER" "$REMOTE_PORT">
                SecurityLevel Sign
                Username "$USER"
                Password "$PASSWORD"
        </Server>
        MaxPacketSize 1500
</Plugin>
```

If link against `libgcrypt` should be there and it is part responsible for encryption in network plugin, then this might be the very reason it fails.

**Comment by mamash** [comment]
Ah OK, confirmed.

**Comment by mamash** [comment]
I fixed this in pkgsrc trunk, will backport to 2016Q4.

**Comment by hadret** [comment]
Great! Many thanks!

**Comment by mamash** [comment]
Should be fixed in 2016Q4, I backported the update to 5.7.1 too.
